src/modules/authorizer/TokenGatedRoleAuthorizer.sol

```diff
@@ -57,29 +65,17 @@ contract TokenGatedRoleAuthorizer is
     //--------------------------------------------------------------------------
     // Overloaded and overriden functions
 
-    /// @inheritdoc RoleAuthorizer
-    /// @dev We add a check to call a different function if the role is token-gated.
-    function isAuthorized(uint8 role, address who)
+    function hasRole(bytes32 roleId, address account)
         public
         view
-        override(RoleAuthorizer, IAuthorizer)
+        virtual
+        override(AccessControlUpgradeable, IAccessControlUpgradeable)
         returns (bool)
     {
-        //Note: since it uses msgSender to generate ID, this should only be used by modules. Users should call hasRole()
-        bytes32 roleId;
-        // If the module self-manages its roles, check if account has the role.
-        if (selfManagedModules[_msgSender()]) {
-            roleId = generateRoleId(_msgSender(), role);
-            //check if token gated:
-            if (isTokenGated[roleId]) {
-                return hasTokenRole(roleId, who);
-            } else {
-                return hasRole(roleId, who);
-            }
-            // If not, check the account against the orchestrator roles
+        if (isTokenGated[roleId]) {
```

Comment:
When you combine this with the onlytokengatedRole from the hasTokenRole then your checking if it is tokenGatedmultiple time in a single function
you might wanna do an internal _hasTokenRole and just use the modifier on the external version